src/System.Management.Automation/utils/MshTraceSource.cs

```diff
@@ -187,7 +187,7 @@ public partial class PSTraceSource
                 // Note, all callers should have already verified the name before calling this
                 // API, so this exception should never be exposed to an end-user.
 
-                throw new ArgumentException("name");
+                throw new ArgumentException(null, nameof(name));
```

Comment:
The same.

Reply:
`name` may be empty not null here.

Reply:
In any case the caller should verify this parameter so it would be appropriate to replace `throw` with `assert`.

Reply:
Same as above.